fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenePostingsTest.java

```diff
@@ -0,0 +1,414 @@
+                // All 3 segments are going to merge to one with all documents
+                assertFieldCountPerSegment(directory, List.of("_0", "_1", "_2", "_3"), List.of(0, 0, 0, 1));
+            }
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_FIELD_METADATA).getCount() >= 4);
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_TERM).getCount() >= 8);
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_POSITIONS).getCount() >= 8);
+            assertTrue(timer.getCounter(LuceneEvents.Waits.WAIT_LUCENE_READ_POSTINGS_FIELD_METADATA).getCount() >= 4);
+            assertTrue(timer.getCounter(LuceneEvents.Waits.WAIT_LUCENE_READ_POSTINGS_TERMS).getCount() >= 3);
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_DELETE_POSTINGS).getCount() >= 3);
+        }
+    }
+
+    @Test
+    void testDeleteAllDocuments() throws Exception {
+        Index index = SIMPLE_TEXT_SUFFIXES_WITH_OPT_POSTINGS;
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, 2.0)
+                .build();
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, "Document 1", 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1623L));
+            recordStore.deleteRecord(Tuple.from(1624L));
+            recordStore.deleteRecord(Tuple.from(1547L));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Map.of());
+            LuceneIndexTestUtils.mergeSegments(recordStore, index);
+            try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                // When deleting all docs from the index, the first segment (_0) was merged away and the last segment (_1) gets removed
+                assertFieldCountPerSegment(directory, List.of("_0", "_1"), List.of(0, 0));
+            }
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_DELETE_POSTINGS).getCount() > 0);
+        }
+    }
+
+    @Test
+    void testComplexDocManyFields() throws Exception {
```

Comment:
I can think of a lot of additional tests that could be beneficial, but given the state of the work, and that this is a temporary pausing point, perhaps it's just worth noting and tracking to come back later.
Mostly, it feels like there's a lot of code that just isn't covered.
1. Tests where positions matter
2. Tests where frequency matters
3. More testing where there are different fields with overlapping terms
4. Tests with terms that may have differences in ordering could mess things up
5. unit test that the ordering is the same between FDB and Lucene terms
6. Tests where impacts matter
7. It might be nice to have some lower-level tests to make sure our api aligns with a built-in implementation, but that may be really hard to get right, because Postings has such a broad api.